### DISCUSSION TITLE
Configurable tab size

### DISCUSSION THREAD

**Discussion by joshtemple** [tconbeer/harlequin#542] [Ideas (Feature Requests)]
Tab size is by default 4 spaces, but it would be useful to be able to configure this to 2 (or another number).

**Comment by joshtemple** [comment]
@tconbeer I was poking around in the codebase trying to find a way to do this. It doesn't seem like the Textual `TextEditor` class supports configuring this, but `TextArea` does. I'm not familiar with Textual, so not clear on why you would choose one or the other. I couldn't even find `TextEditor` in the current version docs for Textual.

Any idea where you would start for this?

**Reply by tconbeer** [reply]
There are a few layers of inheritance here, which makes it confusing. Originally, I had to create my own text area widget, and that was big enough that I spun it off into its own project: https://github.com/tconbeer/textual-textarea. Harlequin uses that project, but then subclasses the widget to add a few harlequin-specific features, like sql formatting.

But then later Textual created an official TextArea widget with most (but not all) of the features we needed, so I updated my https://github.com/tconbeer/textual-textarea project to use the official TextArea widget as its base class.

So the change would start in the https://github.com/tconbeer/textual-textarea project to flow any config down into the built-in TextArea base class. But it's not quite that simple, since I override the `tab` key handler to allow for multi-line indentation (and I think tab-completion), so you'll need to update that. And to be honest, I've seen some parts of the upstream TextArea code that don't look like they fully respect configurable tabs (at least tabs vs spaces), so you'll probably want to do some testing (that also  might be fixed in a recent version of Textual -- that widget has been under very active development).
